src/Lecture3.hs

```diff
@@ -125,10 +134,15 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
-
+    (<>) :: Reward -> Reward -> Reward
+    x <> y = case (rewardSpecial x, rewardSpecial y) of
+        (True, _) -> Reward {rewardGold=rewardGold x <> rewardGold y, rewardSpecial=True}
+        (_, True) -> Reward {rewardGold=rewardGold x <> rewardGold y, rewardSpecial=True}
+        (_, _) -> Reward {rewardGold=rewardGold x <> rewardGold y, rewardSpecial=False}
```

Comment:
You can use the `||` operator on the `rewardSpecial` field to avoid writing verbose pattern-matching